semgrep_metrics.atd

```diff
@@ -57,6 +57,12 @@ type payload = {
      *)
     ~parse_rate <ocaml mutable>: (string (* lang *) * parse_stat) list
                                  <json repr="object">;
+
+    (* Metrics related to osemgrep migration. It's intentionally made optional
+     * so that it's easier to remove the field without breaking backward compatibility
+     * once the migration is complete.
+     *)
+    ?osemgrep <ocaml mutable>: osemgrep_metrics option;
 }
```

Comment:
This change is fine but I don't know why all these fields are marked as mutable. It's rarely necessary.

Reply:
```
commit 2b0b774ee493045b83b6ec6447462e514bc0d83c
Author: Calascibetta Romain <romain.calascibetta@gmail.com>
Date:   Sun Jun 11 12:24:29 2023 +0200

    Set metrics's fields to mutable fields
```
Unfortunately, there's no explanation for this. Please add a note such as `(* TODO: explain why the fields are marked as mutable since this is unusual. *)`

Reply:
I did not know that. I'll try testing them without the mutable fields, and if that works, then I'll merge.

Reply:
And I'll add the comments.